sdk/core/azure-core/test/ut/CMakeLists.txt

```diff
@@ -19,10 +19,15 @@ project (${TARGET_NAME} LANGUAGES CXX)
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
+if(BUILD_TRANSPORT_CURL)
```

Comment:
It would be cleaner if this was its on executable.  Is there a reason you want to conditionally include this into the same executable?